core/src/main/java/org/apache/iceberg/avro/AvroCustomOrderSchemaVisitor.java

```diff
@@ -90,6 +105,10 @@ public T map(Schema map, Supplier<T> value) {
     return null;
   }
 
+  public T variant(Schema variant, Supplier<F> metadataResult, Supplier<F> valueResult) {
```

Comment:
This should function like the map and array cases. It should be `Supplier<T>` rather than `Supplier<F>`.